requests/sessions.py

```diff
@@ -267,6 +277,19 @@ def rebuild_method(self, prepared_request, response):
 
         prepared_request.method = method
 
+    def rewind_body(self, prepared_request):
+        """Move file pointer back to it's recorded starting position
+        so it can be read again on redirect.
+        """
+        body_seek = getattr(prepared_request.body, 'seek', None)
+        if body_seek is not None and isinstance(prepared_request._body_position, int):
+            try:
+                body_seek(prepared_request._body_position)
+            except (IOError, OSError):
+                raise UnrewindableBodyError("An error occured when rewinding request "
+                                            "body for redirect.")
+        else:
+            raise UnrewindableBodyError("Unable to rewind request body for redirect.")
```

Comment:
The variable `self` is unused in this method, which is usually a good sign that it's a function masquerading as a method. I suggest moving this function to utils. =)